edx_rest_framework_extensions/permissions.py

```diff
@@ -52,22 +69,12 @@ class JwtHasScope(BasePermission):
     message = 'JWT missing required scopes.'
 
     def has_permission(self, request, view):
-        if _should_skip_jwt_scope_enforcement(request):
-            return True
         jwt_scopes = decode_jwt_scopes(request.auth)
-        required_scopes = set(self.get_scopes(request, view))
-        if required_scopes.issubset(jwt_scopes):
-            return True
-        return False
-
-    def get_scopes(self, request, view):
-        """
-        Return the required scopes defined on the view.
-        """
-        try:
-            return getattr(view, 'required_scopes')
-        except AttributeError:
-            raise ImproperlyConfigured('JwtHasScope requires the view to define the required_scopes attribute')
+        required_scopes = set(getattr(view, 'required_scopes', []))
+        allowed = bool(required_scopes) and required_scopes.issubset(jwt_scopes)
+        if not allowed:
+            log.warning("Permission JwtHasScope not satisfied.")
```

Comment:
Output of required_scopes and jwt_scopes would be useful, if not computing the missing scope.  Are there other details from the request that one would want if they were looking at these warnings trying to learn something?

Reply:
Done.